src/main/java/com/google/cloud/spanner/pgadapter/utils/MutationBuilder.java

```diff
@@ -0,0 +1,208 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.spanner.pgadapter.utils;
+
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.Mutation.WriteBuilder;
+import com.google.cloud.spanner.jdbc.CloudSpannerJdbcConnection;
+import com.google.cloud.spanner.pgadapter.ConnectionHandler;
+import com.google.spanner.v1.TypeCode;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+public class MutationBuilder {
+
+  private static final long MUTATION_LIMIT = 20000; // PLACEHOLDER
+  private static final long BATCH_LIMIT = 64000; // PLACEHOLDER
+
+  private boolean hasHeader;
+  private boolean isHeaderParsed;
+  private int mutationCount;
+  private int batchSize;
+  private int rowCount;
+  private List<Mutation> mutations;
+  private String tableName;
+  private Map<String, TypeCode> tableColumns;
+  private CSVFormat format;
+  private FileWriter fileWriter;
+
+  public MutationBuilder(
+      String tableName, Map<String, TypeCode> tableColumns, CSVFormat format, boolean hasHeader) {
+    this.mutationCount = 0;
+    this.batchSize = 0;
+    this.hasHeader = hasHeader;
+    this.isHeaderParsed = false;
+    this.tableName = tableName;
+    this.tableColumns = tableColumns;
+    this.format = format;
+    this.mutations = new ArrayList<>();
+  }
+
+  /** @return list of CopyData mutations not yet written into Spanner */
+  public List<Mutation> getMutations() {
```

Comment:
Why do we want a public getter for this? It seems to me like this is internal state of this class.

Reply:
Removed for now.  I don't think this is needed.  Much of this code was original code that the intern wrote during the summer.